File: gptAgentsFE/src/app/modle/agent.ts
export class Agent {
  id?: number;
  role: string;
  description: string;

  constructor(role: string, description: string) {
    this.description = description;
    this.role = role;
  }

  static copy(agent1: Agent, agent2: Agent) {
    agent1.role = agent2.role;
    agent1.description = agent2.description;
  }

  static createFromObj(o: { role: string; description: string }): Agent {
    return new Agent(o.role, o.description);
  }
}

File: backend/gpt-agents/src/main/java/com/example/gptagents/model/Agent.java
package com.example.gptagents.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.antlr.v4.runtime.misc.NotNull;
import org.springframework.lang.NonNull;

@Entity
@Table(name = "Agent")
public class Agent {
    @Id
    @GeneratedValue
    private Long id;

    @NonNull
    private String role;
    @NonNull
    private String description;
    public Agent() {
    }
    public Agent(String role, String description) {
        this.role = role;
        this.description = description;
    }

    public Agent(String agent1) {
        this.role = agent1;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}
